### PR TITLE
Pat/Improve navigation code

### PR DESCRIPTION
**Pull request by whlpatricia**
### Description

Removed set_xxx_context from navigation functions to improve usability.

### Bugzilla bug ID

**Testrail:**
**Link:** https://bugzilla.mozilla.org/show_bug.cgi?id=1938384

### Type of change

Please delete options that are not relevant.

- [ ] New Test
- [ ] New POM
- [x] Other Changes (Stabilization)

### How does this resolve / make progress on that bug?

Completed

### Screenshots / Explanations

NA

### Comments / Concerns

NA

### CODE REVIEW

File: tests/address_bar_and_search/test_adaptive_history_autofill.py [hunk 1]
```diff
@@ -43,29 +43,29 @@ def test_add_adaptive_history_autofill(driver: Firefox):
...
     # Open a new tab, type the first 3 characters of the visited URL
     tabs.new_tab_by_button()
     tabs.wait_for_num_tabs(2)
     driver.switch_to.window(driver.window_handles[-1])
     nav.type_in_awesome_bar("nat")
 
-    autofill_adaptive_element = nav.get_element(
-        "search-result-autofill-adaptive-element"
-    )
+    # tabs.set_chrome_context()
```
Comment (ben-c-at-moz) [inline tests/address_bar_and_search/test_adaptive_history_autofill.py:58]: Seems like these commented lines may need to go?

File: tests/address_bar_and_search/test_suggestion_engine_selection.py [hunk 2]
```diff
@@ -24,6 +24,7 @@ def test_search_suggestion_for_engine_selection(driver: Firefox, site: str):
     nav.type_in_awesome_bar("@")
     nav.element_has_text("results-dropdown", f"Search with {site}")
 
+    nav.set_chrome_context()
```
Comment (ben-c-at-moz) [inline tests/address_bar_and_search/test_suggestion_engine_selection.py:27]: Is there are reason to do this vs. using the context manager?

**Review by ben-c-at-moz — COMMENTED** [review]
A couple things to look at, but other than that, this is great

File: tests/address_bar_and_search/test_suggestion_engine_selection.py (hunk 2 above)
Comment (whlpatricia) [reply tests/address_bar_and_search/test_suggestion_engine_selection.py:27]: the context manager didnt work for some reason

File: tests/address_bar_and_search/test_adaptive_history_autofill.py (hunk 1 above)
Comment (whlpatricia) [reply tests/address_bar_and_search/test_adaptive_history_autofill.py]: oops, uncommented

**Review by ben-c-at-moz — APPROVED** [review]

**Pull request merged**